pyanaconda/modules/boss/install_manager/install_manager.py

```diff
@@ -0,0 +1,264 @@
+        return self._modules
+
+    @available_modules.setter
+    def available_modules(self, modules):
+        """Set available modules which will be used for installation.
+
+        :param modules: Modules list.
+        :type modules: list
+        """
+        self._modules = modules
+
+    def start_installation(self):
+        """Start the installation."""
+        self._collect_tasks()
+
+        self._sum_steps_count()
+        self._disconnect_task()
+        self._tasks_done_step = 0
+        self._installation_terminated = False
+
+        if self._tasks:
+            self._actual_task = self._tasks.pop()
+            self._install_running_signal.emit(True)
+            self._run_task()
+
+    def _collect_tasks(self):
+        self._tasks.clear()
+
+        if not self._modules:
+            log.error("Starting installation without available modules.")
+
+        for module_service in self._modules:
+            # FIXME: This is just a temporary solution.
+            module_object = DBus.get_proxy(module_service, auto_object_path(module_service))
+
+            tasks = module_object.AvailableTasks()
+            for task in tasks:
+                log.debug("Getting task %s from module %s", task[TASK_NAME], module_service)
+                task_proxy = DBus.get_proxy(module_service, task[TASK_PATH])
+                self._tasks.add(task_proxy)
+
+    def _sum_steps_count(self):
+        self._step_sum = 0
+        for task in self._tasks:
+            self._step_sum += task.ProgressStepsCount
+
+    def _run_task(self):
+        if self._installation_terminated:
+            log.debug("Don't run another task. The installation was terminated.")
+            return
+
+        task_name = self._actual_task.Name
+
+        log.debug("Running installation task %s", task_name)
+        self._disconnect_task()
+        self._connect_task()
+        self._task_changed_signal.emit(task_name)
+        self._actual_task.Run()
+
+    def _connect_task(self):
+        s = self._actual_task.ProgressChanged.connect(self._progress_changed)
+        self._subscriptions.append(s)
+
+        s = self._actual_task.RunningChanged.connect(self._task_running_changed)
+        self._subscriptions.append(s)
+
+        s = self._actual_task.ErrorRaised.connect(self._task_error_raised)
+        self._subscriptions.append(s)
+
+    def _disconnect_task(self):
+        for subscription in self._subscriptions:
+            subscription.disconnect()
+
+    def _test_if_running(self, log_msg=None):
+        if self._actual_task is not None:
+            return True
+        else:
+            log.warn(log_msg)
+            return False
+
+    def _task_running_changed(self, is_running):
+        if not is_running:
+            self._tasks_done_step += self._actual_task.ProgressStepsCount
+            if self._tasks:
+                self._actual_task = self._tasks.pop()
+                self._run_task()
```

Comment:
Shouldn't we run next task from a callback of the `Run` method?

Reply:
Based on the previous comment, this is also not true. However, I would prefer signals Started and Done instead of RunningChanged.